netaddr.go

```diff
@@ -805,6 +805,34 @@ func (ip *IP) UnmarshalText(text []byte) error {
 	return err
 }
 
+// MarshalBinary implements the encoding.BinaryMarshaler interface,
+func (ip IP) MarshalBinary() ([]byte, error) {
+	if ip.Is4() {
+		b := ip.As4()
+		return b[:], nil
+	}
+	b := ip.As16()
+	return b[:], nil
+}
+
+// UnmarshalBinary implements the encoding.BinaryUnmarshaler interface.
+func (ip *IP) UnmarshalBinary(b []byte) error {
```

Comment:
Please add check to our fuzz function that does a Marshal/Unmarshal round trip and then checks for identity. This would catch the zone issue and any other edge cases.

Reply:
Is this what you meant?